flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/AsyncExecutionController.java

```diff
@@ -311,6 +318,10 @@ public void drainInflightRecords(int targetNum) {
         }
     }
 
+    public EpochManager getEpochManager() {
+        return epochManager;
```

Comment:
It seems that all usage of this method will be followed by `onNonRecord()`. Thus it might be more straightforward to expose a method like `onNonRecord` in this class. This would also be good for isolating implementation details.

Reply:
👍Thanks for the suggestion, I added `processNonRecord()` for aec, `EpochManager` is not visible for operators now.